azure/scope/machinepoolmachine.go

```diff
@@ -257,7 +270,7 @@ func (s *MachinePoolMachineScope) Close(ctx context.Context) error {
 	)
 	defer done()
 
-	return s.patchHelper.Patch(ctx, s.AzureMachinePoolMachine)
+	return s.PatchObject(ctx)
 }
 
 // UpdateStatus updates the node reference for the machine and other status fields. This func should be called at the
```

Comment:
```go
ctx, _, done := tele.StartSpanWithLogger(
		ctx,
		"scope.MachinePoolMachineScope.Get",
	)
```

Should change "scope.MachinePoolMachineScope.Get" to "scope.MachinePoolMachineScope.UpdateStatus"